prime-dotnet-webapi/Controllers/HealthAuthoritiesController.cs

```diff
@@ -240,5 +246,56 @@ public async Task<ActionResult> UpdatePharmanetAdministratorContacts(int healthA
             await _healthAuthorityService.UpdateContactsAsync<HealthAuthorityPharmanetAdministrator>(healthAuthorityId, contacts);
             return NoContent();
         }
+
+        // GET: api/health-authorities/5/vendors/5/sites
+        /// <summary>
+        /// Checks if the specified vendor is in use by a Health Authority site
+        /// and return the sites that are using the provided vendor
+        /// </summary>
+        /// <param name="healthAuthorityId"></param>
+        /// <param name="vendorId"></param>
+        [HttpGet("{healthAuthorityId}/vendors/{healthAuthorityVendorId}/sites", Name = nameof(GetSitesByVendor))]
+        [Authorize(Roles = Roles.ViewSite)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult> GetSitesByVendor(int healthAuthorityId, int vendorId)
+        {
+            if (!await _healthAuthorityService.HealthAuthorityExistsAsync(healthAuthorityId))
+            {
+                return NotFound($"Health Authority not found with id {healthAuthorityId}");
+            }
+
+            var siteIds = await _healthAuthorityService.GetSitesByVendorAsync(healthAuthorityId, vendorId);
+
+            return Ok(siteIds);
+        }
+
+
+        // GET: api/health-authorities/5/care-types/5/sites
+        /// <summary>
+        /// Checks if the specified care type is in use by a Health Authority site
```

Comment:
Could probably simplify this to
``` returns a list of site ids of a given care type and health authority ```
Since as written it is a little miss leading.

Reply:
Simplified